### PR TITLE
Cut report last message to specified length

### PR DESCRIPTION
**Pull request by K4tsuki**
### Details

Cut last message of a report to specific length, because we only use last message as alternate text on RHN and LHN.

### Fixed Issues

$ https://github.com/Expensify/App/issues/6699

### Tests

1. Send large text message
2. Open developer tools and search for alternateText showed in LHN and RHN on that room.
3. Alternate text should contains trimmed last message and doesn't contain full length of last message
4. Repeat steps above for sending IOU reques, receive new message and receive new IOU message.

### QA Steps

1. Send large text message
2. Open developer tools and search for alternateText showed in LHN and RHN on that room.
3. Alternate text should contains trimmed last message and doesn't contain full length of last message
4. Repeat steps above for sending IOU reques, receive new message and receive new IOU message.

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

#### Web

trim_3

trim_1

trim_2

### CODE REVIEW

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -115,6 +115,17 @@ function getUnreadActionCount(report) {
...
+ * Used before merging Report data.
+ * Cut last message to specific length, because we don't need full last message
+ *
+ * @param {String} lastMessage
+ * @returns {String}
+ */
+function cutLastMessage(lastMessage) {
+    return lastMessage.substr(0, CONST.REPORT.MAX_LAST_MESSAGE_LENGTH);
+}
+
```
Comment (parasharrajat) [inline src/libs/actions/Report.js:128]: Let's move this to reportUits  as `formatReportLastMessageText`

File: src/CONST.js [hunk 2]
```diff
@@ -244,6 +244,7 @@ const CONST = {
         },
         MAX_PREVIEW_AVATARS: 4,
         MAX_ROOM_NAME_LENGTH: 80,
+        MAX_LAST_MESSAGE_LENGTH: 80,
```
Comment (parasharrajat) [inline src/CONST.js:247]:
```suggestion
        LAST_MESSAGE_TEXT_MAX_LENGTH: 80,
```

File: src/libs/actions/Report.js [hunk 3]
```diff
@@ -597,10 +608,12 @@ function updateReportWithNewAction(
     // a chat participant in another application), then the last message text and author needs to be updated as well
     if (newMaxSequenceNumber > initialLastReadSequenceNumber) {
         updatedReportObject.lastMessageTimestamp = reportAction.timestamp;
-        updatedReportObject.lastMessageText = messageText;
         updatedReportObject.lastActorEmail = reportAction.actorEmail;
     }
 
+    // Alwasy cut last message
+    updatedReportObject.lastMessageText = cutLastMessage(messageText);
+
```
Comment (parasharrajat) [inline src/libs/actions/Report.js:616]: Why did you move this out of `if block`?

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/libs/actions/Report.js (hunk 3 above)
Comment (K4tsuki) [reply src/libs/actions/Report.js:616]: There might be a chance that this if block is not executed, so last message is not being cut.

Comment (parasharrajat) [reply src/libs/actions/Report.js:616]: Then it means that message is set somewhere else and we should cut the message there not here.

Comment (K4tsuki) [reply src/libs/actions/Report.js:616]: I have add cuts to possible `lastMessage`, so this is not necessary. Let's remove this.

Comment (parasharrajat) [reply src/libs/actions/Report.js:616]: I see that it is important here. We can remove it. So we can just move the line back to the previous location and add the formatting.

File: src/libs/actions/Report.js (hunk 1 above)
Comment (K4tsuki) [reply src/libs/actions/Report.js]: Done

File: src/libs/actions/Report.js (hunk 3 above)
Comment (K4tsuki) [reply src/libs/actions/Report.js]: Done

File: src/libs/reportUtils.js [hunk 4]
```diff
@@ -220,6 +220,21 @@ function isDeletedAction(action) {
...
+ *
+ * @param {String} lastMessageText
+ * @returns {String}
+ */
+function formatReportLastMessageText(lastMessageText) {
+    if (_.isString(lastMessageText) && lastMessageText.length > CONST.REPORT.LAST_MESSAGE_TEXT_MAX_LENGTH) {
+        return lastMessageText.substr(0, CONST.REPORT.LAST_MESSAGE_TEXT_MAX_LENGTH);
+    }
+
+    return lastMessageText;
```
Comment (parasharrajat) [inline src/libs/reportUtils.js:235]:
```suggestion
    return String(lastMessageText).substring(0, CONST.REPORT.LAST_MESSAGE_TEXT_MAX_LENGTH);
```
image

File: src/libs/reportUtils.js [hunk 5]
```diff
@@ -220,6 +220,21 @@ function isDeletedAction(action) {
     return action.message.length === 0 || action.message[0].html === '';
 }
 
+/**
+ * Used before merging Report data.
+ * For now cut last message text to specific length, because we don't need full last message
+ *
```
Comment (parasharrajat) [inline src/libs/reportUtils.js:226]:
```suggestion
 * Trim the last message text to a fixed limit.
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Testing now...

**Review by parasharrajat — APPROVED** [review]
LGTM.

cc: @Julesssss 
:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by Julesssss — APPROVED** [review]
Tests well for me
